numpy/core/_add_newdocs.py

```diff
@@ -1084,11 +1084,26 @@
 
     Examples
     --------
-    >>> x = np.arange(6).reshape(2,3)
-    >>> np.ascontiguousarray(x, dtype=np.float32)
-    array([[0., 1., 2.],
-           [3., 4., 5.]], dtype=float32)
+    The default memory layout of ndarray is C-contiguous, so
+    in this example `y` just becomes a view of `x`:
+    >>> x = np.arange(6).reshape(2, 3)
+    >>> y = np.ascontiguousarray(x)
+    >>> x.flags['C_CONTIGUOUS']
+    True
+    >>> y.flags['C_CONTIGUOUS']
+    True
+    >>> y.flags['OWNDATA']
+    False
```

Comment:
Thanks, the subtlety here is that `y is x` is actually the same object.  The owndata flag here comes into play because of the reshape!
An option may be to just create the array using `np.ones((2, 3), order="F")` to remove the "clutter" of reshaping and having the order there?

Otherwise, I htink this is great.  I wonder if we have a preferred spelling of `arr.flags.c_contiguous`, but I think we should probably just use this one.

Reply:
Sure, if `.reshape()` is not needed in the examples, we can remove it and simply test `x is y`. Actually, if we use `np.ones()` we will still have the owndata flag:

```
>>> NP = np.ones((2, 3), order="F")
>>> NP.flags
  C_CONTIGUOUS : False
  F_CONTIGUOUS : True
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
```